src/docker-osx-dev

```diff
@@ -829,10 +829,17 @@ function initial_sync {
   log_info "Performing initial sync of paths: ${paths_to_sync[@]}"
 
   local dirs_to_create=()
+  local parent_dirs=()
+  local base_names=()
   local path=""
+  local parent_dir=""
```

Comment:
`parent_dir` and `base_name` could be declared as `local readonly` in the `for` loop below.